src/main/webapp/app/modules/canvas/data-constraints/date-range-component.tsx

```diff
@@ -0,0 +1,150 @@
+import { Content, Picker, Section, Text, TextField, View } from '@adobe/react-spectrum';
+import { Item, Tabs } from '@react-spectrum/tabs';
+import React, { FC, ReactText, useEffect, useRef, useState } from 'react';
+import {
+  formatDate,
+  resetTimezone,
+  startOfDay,
+  strToDate,
+  endOfDay,
+  getStartDateRange,
+  getStartDateRangeInterval,
+  getStartDateRangeTimeUnit,
+  adjustStartDateRangeInterval,
+  stringToDate,
+} from 'app/modules/canvas/data-constraints/utils/date-util';
+import DatePicker from 'app/shared/components/date-picker/date-picker';
+import { DYNAMIC_DATE_RANGE_CONFIG, tabList } from 'app/shared/util/data-constraints.constants';
+
+interface IDateRangeComponentProps {
+  onDateChange: (fromDate, toDate, metadata) => void;
```

Comment:
is it required?, use redux dispatch props and subscribe the date prop in parent function?